frontend/catalyst/programs/verification.py

```diff
@@ -218,9 +263,40 @@ def _obs_checker(obs):
 
     for m in tape.measurements:
         if m.obs:
-            if isinstance(m.obs, Tensor):
-                _ = [_obs_checker(o) for o in m.obs.obs]
-            else:
-                _obs_checker(m.obs)
+            _verify_observable(m.obs, _obs_checker)
+
+    return (tape,), lambda x: x[0]
+
+
+@transform
+def validate_observables(
+    tape: QuantumTape, qjit_capabilities: dict, name: str
+) -> (Sequence[QuantumTape], Callable):
+    """Validates the observables and measurements for a circuit against the capabilites
+    from the TOML file.
+
+    Args:
+        tape (QuantumTape or QNode or Callable): a quantum circuit.
+        qjit_capabilities (dict): specifies the capabilities of the qjitted device
+        name (str): the name of the device to use in error messages.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[.QuantumTape], function]:
+        The unaltered input circuit.
+
+    Raises:
+        CompileError: if an observable is not supported by the device with Catalyst
+
+    """
+
+    def _obs_checker(obs):
+        if not qjit_capabilities.native_obs.get(obs.name):
+            raise CompileError(
+                f"{m.obs} is not supported as an observable on the '{name}' device with Catalyst"
+            )
```

Comment:
#828 Made me realize that we also need to include Catalyst's restrictions in the verifier, just like we do for operations. For instance, Catalyst only supports `{expval, var, probs, counts, sample, state}` and no other measurement processes. There are probably restrictions on observables as well. Additionally, only `expval` and `var` currently support observables, other MPs don't. I guess the 1st and 3rd point should be done in an MP checker, rather than the observable checker, but the 2nd point does seem relevant.

Reply:
I think that should be a separate PR; this one is already decently long, and validation of measurement processes will need to start with determining requirements - we need to figure out what the other restrictions are 😅 So let's let this one be validating that the observables are all supported but the device, and make all MP-related validation another task.

Reply:
MPs for sure, but thank you for adding the observable validation for the catalyst restrictions!